src/fu-engine.c

```diff
@@ -2422,6 +2422,8 @@ fu_engine_install_release(FuEngine *self,
 				      version_rel,
 				      fu_device_get_version(device));
 		fu_device_set_update_error(device, str);
```

Comment:
I'm wondering if we still need to set this -- I'm erring on no. @superm1?

Reply:
I actually think it's worth leaving.  Here's my reasoning:

Outside of chromebook world a mere mortal would do this:
1. `fwupdmgr get-updates`
1. `fwupdmgr update`
2. `fwupdmgr get-devices`

How would they know that step 2 failed if the version isn't discovered until step 3?

Reply:
Or what if they open up a GUI client?  You need to have some way to tell people that the last update didn't work out.

Reply:
> How would they know that step 2 failed if the version isn't discovered until step 3

Wouldn't they get a console error in that case? I can test this a bit in gnome-firmware an fwupdmgr tomorrow -- getting late for today.

Reply:
I tested this upgrading a ColorHug from 2.0.5 to 2.0.6. To simulate the failed update (i.e. where the version was not updated on success) I just did:

	@@ -2413,16 +2413,15 @@ fu_engine_install_release(FuEngine *self,
		/* for online updates, verify the version changed if not a re-install */
		fmt = fu_device_get_version_format(device);
		version_rel = fu_release_get_version(release);
	+       fu_device_set_version(device, "2.0.6");

Then the console output is:

![Screenshot from 2023-12-08 12-39-07](https://github.com/fwupd/fwupd/assets/151380/8cd0f24f-7f25-4dcf-9c2b-26929dd524f9)

and the GUI output is:

![Screenshot from 2023-12-08 12-40-36](https://github.com/fwupd/fwupd/assets/151380/f75a6ddb-68eb-476b-a10d-7924173036c6)

I think we also probably want to use the device update_error for the reason that the firmware can't be deployed, rather than the failure on trying to deploy it. Notably it's also overwritten with a text version of the inhibits/problems in `fu_device_ensure_inhibits()` so I think that's probably two good reasons not to use it :)

Reply:
did you remove the `fu_device_set_update_error()` in your test?

Reply:
Indeed I did!

Reply:
So, waiting @superm1 ? or just remove that from the PR?

Reply:
I did that -- I think it's good now.